app/models/components/system/admin/system_admin_ability_component.rb

```diff
@@ -2,9 +2,7 @@ module System::Admin::SystemAdminAbilityComponent
   include AbilityHost::Component
 
   def define_permissions
-    if user
-      do_not_allow_system_admin_manage_default_instance
-    end
+    do_not_allow_system_admin_manage_default_instance
```

Comment:
we don't use any user property inside, and no one gets to modify the system user?